extensions/kafka-connect/src/test/java/com/hazelcast/jet/kafka/connect/KafkaConnectIntegrationTest.java

```diff
@@ -122,6 +128,51 @@ public void testReading() throws Exception {
         }
     }
 
+    @Test
+    public void test_reading_and_writing_to_map() throws Exception {
+        Properties randomProperties = new Properties();
+        randomProperties.setProperty("name", "datagen-connector");
+        randomProperties.setProperty("connector.class", "io.confluent.kafka.connect.datagen.DatagenConnector");
+        randomProperties.setProperty("max.interval", "1");
+        randomProperties.setProperty("kafka.topic", "orders");
+        randomProperties.setProperty("quickstart", "orders");
+
+        Pipeline pipeline = Pipeline.create();
+        StreamStage<Map.Entry<String, Order>> streamStage = pipeline.readFrom(connect(randomProperties,
+                        rec -> entry(convertToString(rec.keySchema(), rec.key()), new Order(rec))))
```

Comment:
Can we add a test that uses the projection to filter out some items? e.g. odd numbers from the generator.